tests/test-ajax.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Tests for the back-end handling of sticky meta data.
+ *
+ * @package Sticky_Tax
+ */
+
+use LiquidWeb\StickyTax\Ajax as Ajax;
+
+/**
+ * Test suite handling the post meta related to sticky taxonomy posts.
+ */
+class AjaxTest extends WP_Ajax_UnitTestCase {
+
+	public function test_ajax_success() {
+		$_POST['nonce']     = wp_create_nonce( 'sticky-tax' );
+		$_POST['action']    = 'stickytax_get_id_from_name';
+		$_POST['term_name'] = 'Uncategorized';
+		$_POST['term_type'] = 'category';
+
+		try {
+			$this->_handleAjax( 'stickytax_get_id_from_name' );
+		} catch ( WPAjaxDieContinueException $e ) {
+			return;
+		}
+
+		// Check that the exception was thrown.
+		$response = json_decode( $this->_last_response );
+
+		$this->assertInternalType( 'object', $response );
+		$this->assertObjectHasAttribute( 'success', $response );
+		$this->assertTrue( $response->success );
+		$this->assertObjectHasAttribute( 'data', $response );
+	}
+
+	public function test_ajax_error_bad_nonce() {
+		$_POST['nonce']     = uniqid();
+		$_POST['action']    = 'stickytax_get_id_from_name';
+		$_POST['term_name'] = 'Uncategorized';
+		$_POST['term_type'] = 'category';
+
+		try {
+			$this->_handleAjax( 'stickytax_get_id_from_name' );
+		} catch ( WPAjaxDieContinueException $e ) {
+			return;
+		}
+
+		// Check that the exception was thrown.
+		$response = json_decode( $this->_last_response );
+
+		$this->assertInternalType( 'object', $response );
+		$this->assertObjectHasAttribute( 'success', $response );
+		$this->assertFalse( $response->success );
+		$this->assertObjectHasAttribute( 'data', $response );
+		$this->assertObjectHasAttribute( 'errcode', $response->data );
+		$this->assertEquals( $response->data->errcode, 'BAD_NONCE' );
+	}
+
+	public function test_ajax_error_no_term_name() {
```

Comment:
When you have multiple tests that follow the same sort of format, it can be helpful to create a separate method within the test suite to handle a lot of the common setup work. For example, you might add the following:

```php
/**
 * Send a request to the stickytax_get_id_from_name ajax handler.
 *
 * @param string $term_name The term name to submit.
 * @param string $term_type The taxonomy to which the term belongs.
 * @return object The JSON-decoded ajax response.
 */
protected function make_ajax_request( $term_name, $term_type ) {
  $_POST = [
    'nonce'     => wp_create_nonce( 'sticky-tax' ),
    'action'    => 'stickytax_get_id_from_name',
    'term_name' => $term_name,
    'term_type' => $term_type,
  ];

  $this->_handleAjax( 'stickytax_get_id_from_name' );

  return json_decode( $this->_last_response );
}
```

Each of the other tests would then become a matter of:

```php
public function test_ajax_error_no_term_name() {
  $response = $this->make_ajax_request( '', 'category' );

  $this->assertFalse( $response->success );
  $this->assertEquals( 'NO_TERM_NAME', $response->data->errcode );
}
```

This makes each test much clearer about what is actually being tested, and what is just setup.

Reply:
can you confirm this works in testing? When I do it, it fails on confirming object type.

Reply:
I've pushed an update to this branch that simplifies the tests in two ways:

1. Introduce the `make_ajax_request()` method, which accepts an array of arguments to merge into some defaults (that way, the test case only needs to specify which overrides to make).
2. Further simplify the `test_ajax_error_*` methods by using a [PHPUnit data provider](https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers), so we only have to write the test logic once.